packages/theme/composables/useApi/index.ts

```diff
@@ -1,14 +1,82 @@
 import { useContext } from '@nuxtjs/composition-api';
 import type { Variables } from 'graphql-request';
+import type { UseApiInterface } from './UseApi';
 
-type Request = <T = any, V = Variables>(document: string, variables?: V, requestHeaders?: HeadersInit) => Promise<T>;
-
-export const useApi = () => {
+/**
+ * The `useApi()` composable allows executing GraphQL queries and mutations.
+ *
+ * @example
+ *
+ * Execute a GraphQL Query and handle its error, loading and result:
+ *
+ * ```ts
+ * import { ref, readonly } from '@nuxtjs/composition-api';
+ * import { useApi } from '~/composables/useApi';
+ *
+ * const GET_CART_PRICE_QUERY = `
+ *   query GET_CART_PRICE_QUERY($cartId: String!) {
+ *     cart(cart_id: $cartId) {
+ *       prices {
+ *         subtotal_excluding_tax {
+ *           value
+ *         }
+ *       }
+ *     }
+ *   }
+ * `;
+ *
+ * export default {
+ *   setup() {
+ *     const { query } = useApi();
+ *
+ *     const price = ref<number | null>(null);
+ *     const loading = ref(false);
+ *     const error = ref<Error | null>(null);
+ *
+ *     const getCartPrice = async (cartId: string) => {
+ *       loading.value = false;
+ *       try {
+ *         const result = await query(GET_CART_PRICE_QUERY, { cartId }, {
```

Comment:
```suggestion
 *         const result = await query<SomeDummyType>(GET_CART_PRICE_QUERY, { cartId }, {
```
May be worth demonstrating that passing a generic is possible (don't treat the suggestion verbatim, I'm just not sure how to name SomeDummyType :P)